api.planx.uk/modules/flows/validate/service/templatedNodes.ts

```diff
@@ -12,33 +12,51 @@ const validateTemplatedNodes = async (
   flowId: string,
   flowGraph: FlowGraph,
 ): Promise<FlowValidationResponse> => {
-  const templatedFlowEdits = await getTemplatedFlowEdits(flowId);
-
-  if (isNull(templatedFlowEdits.templatedFrom)) {
+  const primaryTemplatedFlowEdits = await getTemplatedFlowEdits(flowId);
+  if (isNull(primaryTemplatedFlowEdits.templatedFrom)) {
     return {
       title: "Templated nodes",
       status: "Not applicable",
       message: "This is not a templated flow",
     };
   }
 
+  // Because we have a flattened flow here, we'll need to get templated flow edits for this flow id and any nested templated flow IDs
+  const nestedTemplatedFlowIds = Object.entries(flowGraph as Flow["data"])
+    .filter(
+      ([_nodeId, node]) =>
+        node.type === ComponentType.InternalPortal &&
+        node.data?.flattenedFromExternalPortal === true &&
+        Boolean(node.data?.templatedFrom),
+    )
+    .map(([nodeId, _node]) => nodeId);
+
+  const editedNestedTemplatedNodeIds = await Promise.all(
+    nestedTemplatedFlowIds.map((nestedFlowId) =>
+      getTemplatedFlowEdits(nestedFlowId),
+    ),
+  ).then((values) => {
+    return values.flatMap((value) => Object.keys(value.edits?.data || {}));
+  });
+
+  const allEditedTemplatedNodeIds = Object.keys(
+    primaryTemplatedFlowEdits.edits?.data || {},
+  ).concat(editedNestedTemplatedNodeIds);
```

Comment:
This is the most important change ! 

When we get `allRequiredTemplatedNodeIds` we iterate over the _flattened_ structure, which naturally picks up required templated nodes within nested templated flows.

But when calculating `allEditedTemplatedNodes`, we were previously only fetching _this_ flows 'edits'/'customisations'. We need to fetch the edits for this flow _and each nested templated flow_ before comparing if every required templated node has been edited :heavy_plus_sign: